runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractContextStreamView.java

```diff
@@ -257,7 +249,8 @@ public boolean hasNext() {
         final List<ILogData> dataList = new ArrayList<>();
         ILogData thisData;
 
-        while ((thisData = getNextEntry(context, maxGlobal)) != null) {
+        while ((thisData = streamReadWithCheck(()->getNextEntry(context, maxGlobal), maxGlobal)) != null) {
```

Comment:
This getNextEntries is the super class impl, it has a overridden impl. You need to change that.